src/py_eddy_tracker/dataset/grid.py

```diff
@@ -849,42 +852,59 @@ def eddy_identification(
                     obs.amplitude[:] = amp.amplitude
                     obs.speed_average[:] = max_average_speed
                     obs.num_point_e[:] = contour.lon.shape[0]
-                    xy_e = resample(contour.lon, contour.lat, **out_sampling)
-                    obs.contour_lon_e[:], obs.contour_lat_e[:] = xy_e
                     obs.num_point_s[:] = speed_contour.lon.shape[0]
-                    xy_s = resample(
-                        speed_contour.lon, speed_contour.lat, **out_sampling
-                    )
-                    obs.contour_lon_s[:], obs.contour_lat_s[:] = xy_s
 
-                    # FIXME : we use a contour without resampling
-                    # First, get position based on innermost contour
-                    centlon_i, centlat_i, _, _ = _fit_circle_path(
-                        create_vertice(inner_contour.lon, inner_contour.lat)
+                    # Evenly resample contours with nb_pts = nb_pts_original x presampling_multiplier
+                    xy_i = uniform_resample(
+                        inner_contour.lon,
+                        inner_contour.lat,
+                        fixed_size=inner_contour.lon.shape[0] * presampling_multiplier,
```

Comment:
you must use num_fac option instead of fixed_size : https://github.com/AntSimi/py-eddy-tracker/blob/92c326230f7a9ca8c53147f78890a8c7321de8f5/src/py_eddy_tracker/generic.py#L305